internal/test/smoke/size_limits_test.go

```diff
@@ -0,0 +1,250 @@
+		defer resp.Body.Close()
+	}
+	require.Equal(t, http.StatusBadRequest, resp.StatusCode)
+	errCurrentBytes, err := io.ReadAll(resp.Body)
+	require.NoError(t, err)
+	errCurrent := &public.ErrorResponse{}
+	err = json.Unmarshal(errCurrentBytes, errCurrent)
+	require.NoError(t, err)
+	require.NotNil(t, errCurrent.Errors)
+	require.NotEmpty(t, *errCurrent.Errors)
+	require.Equal(t, strconv.Itoa(http.StatusBadRequest), (*errCurrent.Errors)[0].Status)
+	regExp := `request body has an error: reading failed: read tcp (127\.0\.0\.1|\[::1\]):(\d+)->(127\.0\.0\.1|\[::1\]):(\d+): (i/o timeout)`
+	require.Regexp(t, regExp, (*errCurrent.Errors)[0].Title)
+}
+
+func (s *SuiteLimitSize) TestIdleTimeout() {
+	t := s.T()
+	// TODO Pending test to check idle timeout
+	//      Initial idea was to do a request with
+	//      Connection: Keep-Alive
+	//      And await the time for idle timeout
+	//      and check in some way if the connection
+	//      was still open; but I have not found
+	//      yet a way to check this; it was observed
+	//      that even sending the Connection: Keep-Alive
+	//      the reponse was not sending back the header
+	//      with any value (indicating if the connection
+	//      is immeditly closed or similar).
+	t.SkipNow()
+}
+
+func (s *SuiteLimitSize) TestWriteTimeout() {
+	t := s.T()
+	// TODO Skiping so far, but open for ideas about check this behavior
+	t.SkipNow()
+}
+
+func (s *SuiteLimitSize) TestLimitSizeRequestHeader() {
+	requestID := "test_limit_size_request_header"
+	t := s.T()
+
+	// Create a HTTP client and sent the request
+	client := &http.Client{}
+
+	hdr := http.Header{}
+	url := s.DefaultPublicBaseURL() + "/domains/token"
+	s.addRequestID(&hdr, requestID)
+	reqBody, err := json.Marshal(&public.DomainRegTokenRequest{
+		DomainType: public.RhelIdm,
+	})
+	require.NoError(t, err)
+	require.NotEmpty(t, reqBody)
+	reqBodyReader := bytes.NewReader(reqBody)
+	req, err := http.NewRequestWithContext(
+		context.Background(),
+		http.MethodPost,
+		url,
+		reqBodyReader,
+	)
+	require.NoError(t, err)
+	require.NotNil(t, req)
+	req.Header.Set(echo.HeaderContentType, echo.MIMEApplicationJSON)
+	for key, value := range hdr {
+		req.Header.Set(key, strings.Join(value, "; "))
+	}
+	// Override
+	s.currentXRHID = &s.userXRHID
+	s.addXRHIDHeader(&req.Header, s.currentXRHID)
+
+	// Header limit is triky to check, because even if we set
+	// the value of 1 byte, the internal server set that size
+	// plus 4096 bytes, see link below:
+	// https://cs.opensource.google/go/go/+/master:src/net/http/server.go;l=926
+	// so we have to go over that threshould on the headers to
+	// get an 431 too large error response for the headers.
+	bufHeader := strings.Builder{}
+	require.NoError(t, req.Header.Write(&bufHeader))
+	bufHeaderString := bufHeader.String()
+	limit := 4096 + s.Config.Application.SizeLimitRequestHeader
+	if len(bufHeaderString) <= limit {
+		req.Header.Add("X-Padding", strings.Repeat("-", limit-len("X-Padding")))
```

Comment:
Is this "precise" calculation needed?  I think it is good enough to get the error when it is definitely more than expected. 

The whole test could be shorter:
```func (s *SuiteLimitSize) TestLimitSizeRequestHeader() {
	t := s.T()

	// Create a HTTP client and sent the request
	client := &http.Client{}

	url := s.DefaultPublicBaseURL() + "/domains/token"
	reqBody, err := json.Marshal(&public.DomainRegTokenRequest{
		DomainType: public.RhelIdm,
	})
	require.NoError(t, err)
	require.NotEmpty(t, reqBody)
	reqBodyReader := bytes.NewReader(reqBody)
	req, err := http.NewRequestWithContext(
		context.Background(),
		http.MethodPost,
		url,
		reqBodyReader,
	)
	require.NoError(t, err)
	require.NotNil(t, req)

	// Header limit is tricky to check, because even if we set
	// the value of 1 byte, the internal server set that size
	// plus 4096 bytes, see link below:
	// https://cs.opensource.google/go/go/+/master:src/net/http/server.go;l=926
	// so we have to go over that threshold on the headers to
	// get an 431 too large error response for the headers.
	headerSize := 4096 + s.Config.Application.SizeLimitRequestHeader
	tooBigHeader := strings.Repeat("-", headerSize)
	req.Header.Add("X-Padding", tooBigHeader)

	resp, err := client.Do(req)
	if resp != nil {
		defer resp.Body.Close()
	}
	require.NoError(t, err)
	require.NotNil(t, resp)
	require.Equal(t, http.StatusRequestHeaderFieldsTooLarge, resp.StatusCode)
}
```

Btw there are 2 typos fixed in the comment (threshold, tricky)